src/vma/proto/dst_entry.cpp

```diff
@@ -251,6 +265,9 @@ bool dst_entry::resolve_net_dev(bool is_connect)
 
 	if (update_rt_val()) {
 		ret_val = update_net_dev_val();
+		if (ret_val) {
+			set_src_addr();
```

Comment:
In face setting source address is not directly related to net_dev_val and to move it to update_net_dev_val it will appear that it depend on this function given that it depend route resolving and update_rt_val() as well.